### PR TITLE
Verify number of taskmanagers reported by Flink dashboard when monitoring

### PR DESCRIPTION
**Pull request by Bronek**
This will add HTTP call to Flink dashboard inside the monitoring job, and in turn compare the number of taskmanagers (as reported by Flink) against the expected number. Since this call makes use of a significant part of the stack, it is a good end-to-end test.

Extended the unit test for the "happy path", yet need to extend unit tests to cover error cases (e.g. dashboard not reachable etc.)

Note, as a result of refactoring there is also an internal change in how dashboard URL is reported by `paasta status` because now annotation will contain the full URL rather than only "root" part.

### CODE REVIEW

File: paasta_tools/check_flink_services_health.py [hunk 1]
```diff
@@ -53,6 +57,66 @@ def healthy_flink_containers_cnt(si_pods: Sequence[V1Pod], container_type: str)
...
+            "Reasons this might be happening:\n"
+            "\n"
+            "  The service may simply be unhealthy. There also may not be enough resources\n"
+            "  in the cluster to support the requested instance count.\n"
+            "\n"
+            "Things you can do:\n"
+            "\n"
+            "  * Fix the cause of the unhealthy service. Try running:\n"
+            "\n"
+            "      paasta status -s %(service)s -i %(instance)s -c %(cluster)s -vv\n"
```
Comment (vkhromov) [inline paasta_tools/check_flink_services_health.py]: this is probably copy-pasted from somewhere, but a multiline string would be much more readable here.

File: paasta_tools/check_flink_services_health.py [hunk 2]
```diff
@@ -70,7 +134,22 @@ def check_flink_service_health(
...
+        )
+        reported_taskmanagers = overview.get("taskmanagers", 0)
+    except ValueError as e:
+        strerror = str(e)
+
+    send_event_if_not_enough_taskmanagers(
+        instance_config=instance_config,
+        expected_count=taskmanagers_expected_cnt,
+        num_reported=reported_taskmanagers,
+        strerror=strerror,
```
Comment (vkhromov) [inline paasta_tools/check_flink_services_health.py]: It's actually surprising that `mypy` says nothing here.
`strerror` and `reported_taskmanagers` can be `None`, but are passed into a function expecting `str` and `int` types accordingly.

File: paasta_tools/flink_tools.py [hunk 3]
```diff
@@ -217,5 +220,29 @@ def set_flink_desired_state(
...
+    try:
+        response = _dashboard_get(service, instance, cluster, "overview")
+        return json.loads(response)
+    except requests.RequestException as e:
+        url = e.request.url
+        err = e.strerror
+        raise ValueError(f"failed HTTP request to Jobmanager dashboard {url}: {err}")
+    except json.JSONDecodeError as e:
+        err = str(e)
+        raise ValueError(f"JSON decoding error from Jobmanager dashboard: {err}")
```
Comment (vkhromov) [inline paasta_tools/flink_tools.py]: why not just `{e}` instead of `{str(e)}`?

File: paasta_tools/check_flink_services_health.py (hunk 2 above)
Comment (Bronek) [reply paasta_tools/check_flink_services_health.py]: Thank you, I added `Optional` here. Indeed it is surprising that mypy missed it.

**Pull request merged**